### ISSUE TITLE
Access always denied for images with .JPG extension

### ISSUE THREAD

**Issue by alexmogavero** [yonjah/piwigo-privacy#11]
Hi,
I am trying to use this plugin, but in my installation (using Apache) I cannot make it work.
I have followed your guide page using the "naive" approach, but I loose access to almost all my pictures with the exception of few albums. I cannot really understand what those album have different than the others.

Probably I am missing some configuration steps? If so could it be possible to have some more detailed usage info?

Thank you!

**Comment by yonjah** [comment]
Hi @alexmogavero 
I'm happy to accept pull request / wiki suggestion with what ever instructions you think are required.
I haven't used apache in years so I don't have a working configuration but using the naive mode should work regardless of the server you use.

To help you debug this issue please attach a few sample links of urls that work or doesn't work ( You can replace the domain with sample.com and any digit with the digit 1 ) see if there is any obvious difference in their format.

Previous versions had issue with images that were uploaded through ftp but that should be fixed.
Are you using any plugins who might change the way images are uploaded ?

do you have access to php error log ?

**Comment by alexmogavero** [comment]
This is a link that works: https://sample.com/photos/plugins/piwigo_privacy/get.php?img_id=11111&file=_data/i/galleries/Album/SubAlbum/Name_111-me.jpg

While this doesn't
https://sample.com/photos/plugins/piwigo_privacy/get.php?img_id=11111&file=_data/i/galleries/Album/DSC_111-me.JPG

I upload images manually via ssh on the galleries folder
I have access to everything, though I am not an expert of php, so I am not sure how to get the log.

**Comment by alexmogavero** [comment]
I tried but I cannot locate the log file for your plugin.
However if I copy-paste the address of the images I cannot see into the browser I get the "access denied" error.
Moreover I have noticed that the main difference between the images that are visible and those that are not is the extension: .jpg are visible .JPG are not visible.
Can this be the issue?!

**Comment by alexmogavero** [comment]
I have done a test and the followings are the steps to reproduce the bug (or at least I think it is a bug):

1. Create an album with one image with extension `.jpg`
2. Create another album copy-pasting the album above and changing the extension to `.JPG`
3. While the first album will work correctly the second one will not show any picture giving access denied messages

It seems therefore that this pluging cannot handle files with extension .JPG, so it seems to be case sensitive on the extension while it should not.
I would change the title of this issue and flag it as a bug.

**Comment by alexmogavero** [comment]
I managed to get an error log:
```
Fatal error: Uncaught exception 'Exception' with message 'File extension is not allowed' in /var/www/html/photos/plugins/piwigo_privacy/helper_funcs.inc.php:25
Stack trace:
#0 /var/www/html/photos/plugins/piwigo_privacy/helper_funcs.inc.php(40): pwg_privacy_error('File extension ...')
#1 /var/www/html/photos/plugins/piwigo_privacy/helper_funcs.inc.php(60): pwg_privacy_reject_access('File extension ...')
#2 /var/www/html/photos/plugins/piwigo_privacy/get.php(69): pwg_privacy_serve_file('_data/i/./galle...')
#3 {main} thrown in /var/www/html/photos/plugins/piwigo_privacy/helper_funcs.inc.php on line 25
```

This obviously confirms my hypothesis of a problem with the file extension.
From the stack trace I figured that the error generates at `helper_funcs.inc.php(60)`. Here a bit of context:
```php
	$ext = get_extension($path);

	if (!in_array($ext, $conf['file_ext'])) {
		pwg_privacy_reject_access('File extension is not allowed');  // <----- line 60
	}

	$range_support = false;

	$ext = strtolower($ext);
	switch ($ext) {
		case 'jpe': case 'jpeg': case 'jpg':
			$mime='image/jpeg';
			break;
```

I guess that moving the line `$ext = strtolower($ext);` soon after `$ext = get_extension($path);` will solve the issue.

**Comment by yonjah** [comment]
Sorry for taking so long to address this issue.
Merged your fix in 0.1.4

**Comment by alexmogavero** [comment]
Thank you!
